runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactStagingServiceTest.java

```diff
@@ -0,0 +1,271 @@
+        public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+          Files.deleteIfExists(dir);
+          return FileVisitResult.CONTINUE;
+        }
+      });
+    }
+  }
+
+  private void putArtifact(String stagingSessionToken, final String filePath, final String fileName)
+      throws Exception {
+    StreamObserver<PutArtifactRequest> outputStreamObserver = stub
+        .putArtifact(new StreamObserver<PutArtifactResponse>() {
+          @Override
+          public void onNext(PutArtifactResponse putArtifactResponse) {
+            Assert.fail("OnNext should never be called.");
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            throwable.printStackTrace();
+            Assert.fail("OnError should never be called.");
+          }
+
+          @Override
+          public void onCompleted() {
+          }
+        });
+    outputStreamObserver.onNext(PutArtifactRequest.newBuilder().setMetadata(
+        PutArtifactMetadata.newBuilder()
+            .setMetadata(ArtifactMetadata.newBuilder().setName(fileName).build())
+            .setStagingSessionToken(stagingSessionToken)).build());
+
+    FileInputStream fileInputStream = new FileInputStream(filePath);
+    byte[] buffer = new byte[1 << 20];
+    int read = fileInputStream.read(buffer);
+    while (read != -1) {
+      outputStreamObserver.onNext(PutArtifactRequest.newBuilder().setData(
+          ArtifactChunk.newBuilder().setData(ByteString.copyFrom(buffer, 0, read)).build())
+          .build());
+      read = fileInputStream.read(buffer);
+    }
+    outputStreamObserver.onCompleted();
+  }
+
+  private String commitManifest(String stagingSessionToken, List<ArtifactMetadata> artifacts)
+      throws Exception {
+    CompletableFuture<String> stagingTokenFuture = new CompletableFuture<>();
+    stub.commitManifest(
+        CommitManifestRequest.newBuilder().setStagingSessionToken(stagingSessionToken)
+            .setManifest(Manifest.newBuilder().addAllArtifact(artifacts).build()).build(),
+        new StreamObserver<CommitManifestResponse>() {
+          @Override
+          public void onNext(CommitManifestResponse commitManifestResponse) {
+            stagingTokenFuture.complete(commitManifestResponse.getStagingToken());
+          }
+
+          @Override
+          public void onError(Throwable throwable) {
+            Assert.fail("OnError should never be called.");
+          }
+
+          @Override
+          public void onCompleted() {
+
+          }
+        });
+    return stagingTokenFuture.get(1, TimeUnit.SECONDS);
+
+  }
+
+  @Test
+  public void generateStagingSessionTokenTest() throws Exception {
+    String basePath = destDir.toAbsolutePath().toString();
+    String stagingToken = artifactStagingService.generateStagingSessionToken("abc123", basePath);
+    Assert.assertEquals(
+        "{\"sessionId\":\"abc123\",\"basePath\":\"" + basePath + "\"}", stagingToken);
```

Comment:
This looks like a change detector test but I will still like to keep it to make sure that we don't change the token format by mistake.